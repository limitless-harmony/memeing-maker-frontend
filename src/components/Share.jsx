import React from 'react';
import styled from 'styled-components';

import {
  FacebookIcon,
  TwitterIcon,
  LinkedinIcon,
  WhatsAppIcon,
} from 'components/share-icons';
import { calculateRem } from 'styles';
import ShareButtons from 'components/ShareButton';

const Share = () => (
  <StyledShare>
    <IconSection>
      <ShareButtons network="facebook">
        <FacebookIcon round />
      </ShareButtons>
      <ShareButtons network="twitter">
        <TwitterIcon round />
      </ShareButtons>
      <ShareButtons network="whatsapp">
        <WhatsAppIcon round />
      </ShareButtons>
      <ShareButtons network="linkedin">
        <LinkedinIcon round />
      </ShareButtons>
    </IconSection>
  </StyledShare>
);

const StyledShare = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  text-align: center;
`;

const IconSection = styled.div`
  display: flex;
  width: 100%;
  align-items: center;
  padding: ${calculateRem(20)} ${calculateRem(40)};
  min-height: ${calculateRem(80)};
  box-sizing: border-box;
`;

export default Share;
