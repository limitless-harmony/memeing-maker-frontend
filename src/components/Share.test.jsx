import React from 'react';
import { shallow } from 'enzyme';
import Share from 'components/Share';

describe('Testing the Share component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Share />);
    expect(wrapper.find('div'));
    expect(wrapper).toMatchSnapshot();
  });
});
