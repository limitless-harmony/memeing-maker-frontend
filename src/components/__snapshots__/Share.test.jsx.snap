// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Share component renders without crashing 1`] = `
<styled.div>
  <styled.div>
    <Connect(ShareButton)
      network="facebook"
    >
      <Icon
        logoFillColor="white"
        round={true}
        size={40}
      />
    </Connect(ShareButton)>
    <Connect(ShareButton)
      network="twitter"
    >
      <Icon
        logoFillColor="white"
        round={true}
        size={40}
      />
    </Connect(ShareButton)>
    <Connect(ShareButton)
      network="whatsapp"
    >
      <Icon
        logoFillColor="white"
        round={true}
        size={40}
      />
    </Connect(ShareButton)>
    <Connect(ShareButton)
      network="linkedin"
    >
      <Icon
        logoFillColor="white"
        round={true}
        size={40}
      />
    </Connect(ShareButton)>
  </styled.div>
</styled.div>
`;
